src/hooks/useResponsive.ts

```diff
@@ -0,0 +1,71 @@
+import { useEffect, useState } from 'react'
+import { matchMediaQuery as defaultMatchMediaQuery } from '@styles'
+
+type Device = 'desktop' | 'tablet' | 'mobile'
+type MaxWidth = number
+
+type ResponsiveCase<T, K extends keyof T> = {
+  [key in Device]: T[K]
+}
+type MatchMediaQuery = {
+  [key in Device]: `(max-width: ${MaxWidth}px)`
+}
+type MatchMediaQueryParams = MatchMediaQuery | undefined
+
+/**
+ *
+ * @example
+ * const avatarSize = useResponsive<AvatarProps, 'size'>({
+ *   desktop: 'medium',
+ *   tablet: 'small',
+ *   mobile: 'small'
+ * });
+ */
+export const useResponsive = <T, K extends keyof T>(
+  responsiveCase: ResponsiveCase<T, K>,
+  matchMediaQuery: MatchMediaQueryParams = defaultMatchMediaQuery as MatchMediaQuery
+): T[K] => {
+  const getDevice = (queries: MatchMediaQuery): Device => {
+    const defaultDevice: Device = 'desktop'
+
+    if (typeof window === 'undefined') {
+      return defaultDevice
+    }
+
+    return Object.keys(queries).reduce<Device>((prev, cur) => {
+      const device = cur as Device
+      const query = queries[device] as string
+
+      return window.matchMedia(query).matches ? device : prev
+    }, defaultDevice)
```

Comment:
```suggestion
    return (Object.keys(queries) as Device[]).reduce<Device>((prev, cur) => {
      const device = cur
      const query = queries[device]

      return window.matchMedia(query).matches ? device : prev
    }, defaultDevice)
```

요렇게 수정하면 불필요한 단언인 as string을 제거할 수 있을 것 같습니다 !

Reply:
[6417182](https://github.com/price-offer/offer-fe/pull/53/commits/6417182f5f0d81713dd0dd33c593383f623f527b)에서 적용 하였슴니다! 감사합니당!